der/src/ord.rs

```diff
@@ -11,27 +11,36 @@ use core::{cmp::Ordering, marker::PhantomData};
 /// This is used by the DER encoding for `SET OF` in order to establish an
 /// ordering for the elements of sets.
 pub trait DerOrd {
+    /// An error returned when the comparison fails.
+    type Error: From<Error>;
```

Comment:
Per my comments on #1053, I think things would be simpler without this associated type, keeping the error type as `der::Error`, and leveraging the `From<der::Error>` bound in e.g. `Decode`/`Encode` to handle converting error types.

Same goes for `ValueOrd`.

Reply:
Fixed this.